### ISSUE TITLE
Backorders+Preorders > availability date cannot be set anymore > regression

### ISSUE THREAD

**Issue by Marc-pi** [woocommerce/google-listings-and-ads#1796]
### Describe the bug:

Availabilty date cannot be set for backorders, which leads to product feed error
![image](https://user-images.githubusercontent.com/919332/204352464-35fa14c0-3050-4cd0-a855-6e1035212199.png)

### Steps to reproduce:
1. CASE 1 - BACKORDER - Create a single product with Manage Stock set to ON (checkbox)
     * Set the dropdown to Allow Backorders for that product 
     * Make sure it has a negative or zero stock quantity
2. CASE 2 - PREORDER Create a single product with Manage Stock set to OFF (checkbox)
3. Check the GLA tab where there should be an availability date field

### Expected behavior:
The field should be visible, it is currently not
As long as i understand, it was a date to set, but in case of preorder, there is no way to set a duration (ex:  need 3 days to give back the ordered quantity).

### Actual behavior:
the field is currently not visible, it is a regression
in case of preorder, there is not dedicated field (or auto calculated field deducted from a duration)

### Additional details:
Woocommerce 7.1 / Woo Blocks 9.0 / GLA 2.3.1
Related to https://github.com/woocommerce/google-listings-and-ads/pull/951

**Comment by puntope** [comment]
Howdy @Marc-pi 

I was unable to reproduce this issue. 

**Steps I followed for Manage Stock set to ON:**

1. Manage Stock set to ON
2. Allow Backorders
3. Set stock to 0
4. Go to GLA Attributes and set Availability Date and time
5. Sync

**Steps I followed for Manage Stock set to OFF:**

1. Manage Stock set to OFF
4. Go to GLA Attributes and set Availability Date and time
5. Sync

**Result**

<img width="1147" alt="Screenshot 2022-11-29 at 14 27 31" src="https://user-images.githubusercontent.com/5908855/204455412-078f2cd0-3fa0-4296-8ba5-3c5fc66e60f4.png">
